src/messages/mod.rs

```diff
@@ -743,6 +770,69 @@ impl Debug for DirectMessage {
             ProxyRateLimitExceeded { ref ack } => {
                 write!(formatter, "ProxyRateLimitExceeded({:?})", ack)
             }
+            Vote { ref vote } => write!(formatter, "{:?}", vote),
+            Block { ref block } => write!(formatter, "{:?}", block),
+            ChainRequest => write!(formatter, "ChainRequest"),
+            ChainResponse {
+                ref blocks,
+                ref pending_blocks,
+            } => {
+                write!(
+                    formatter,
+                    "ChainResponse {{ blocks: {:?}, pending_blocks: {:?} }}",
+                    blocks,
+                    pending_blocks
+                )
+            }
+            NodeApproval {
+                ref blocks,
+                ref pending_blocks,
+                ref non_elders,
+                ref not_yet_valid_blocks,
+            } => {
+                write!(
+                    formatter,
+                    "NodeApproval {{ blocks: {:?}, pending_blocks: {:?}, non_elders: {:?}, \
+                    not_yet_valid_blocks: {:?} }}",
+                    blocks,
+                    pending_blocks,
+                    non_elders,
+                    not_yet_valid_blocks
+                )
+            }
+            EncryptionKeyRequest { ref public_sign_key } => {
+                write!(
+                    formatter,
+                    "EncryptionKeyRequest {{ {:02x}{:02x}{:02x}.. }}",
+                    public_sign_key.0[0],
+                    public_sign_key.0[1],
+                    public_sign_key.0[2]
+                )
+            }
+            EncryptionKeyResponse {
+                ref public_sign_key,
+                ref public_encrypt_key,
+            } => {
+                write!(
+                    formatter,
+                    "EncryptionKeyResponse {{ public_sign_key: {:02x}{:02x}{:02x}.., \
```

Comment:
the encryption key belongs to which public_sign_key is not printed out?
this will be difficult to tracing the message flow.

Reply:
The encryption key is printed out too if it's not `None`.